src/gui/dialogs/statistics_dialog.cpp

```diff
@@ -133,34 +151,169 @@ void statistics_dialog::add_damage_row(
 	const long long dsa = shift * damage      - expected;
 	const long long dst = shift * turn_damage - turn_expected;
 
-	const long long shifted = ((expected * 20) + shift) / (2 * shift);
-	std::ostringstream str;
-	str << damage << " / "
-		<< static_cast<double>(shifted) * 0.1
-		<< "    " // TODO: should probably make this two columns
-		<< (((dsa < 0) ^ (expected < 0)) ? "" : "+")
-		<< (expected == 0 ? 0 : 100 * dsa / expected) << '%';
-
-	item["label"] = str.str();
+	const auto& damage_str = [shift](long long damage, long long expected) {
```

Comment:
Shouldn't be a reference. I'm not even sure what this would technically reference...

Reply:
`damage_str` would reference an anonymous lambda function that exists until the end of the block. https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary

I've removed the `&`, but I think the code is basically equivalent with and without it.